File: leetcode/editor/en/[2231]Largest Number After Digit Swaps by Parity.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {number} num
 * @return {number}
 */
var largestInteger = function (num) {
    let odd = new MinPriorityQueue();
    let even = new MinPriorityQueue();
    let reference = num;
    let multiply = 1;
    let result = 0;

    while (num > 0) {
        const temp = num % 10;
        if (temp % 2 === 1) {
            odd.enqueue(temp);
        } else {
            even.enqueue(temp);
        }

        num = Math.floor(num / 10);
    }

    while (reference > 0) {
        const temp = reference % 10;

        if (temp % 2 === 1) {
            result = odd.dequeue().element * multiply + result;
        } else {
            result = even.dequeue().element * multiply + result;
        }

        reference = Math.floor(reference / 10);
        multiply *= 10;
    }

    return result;
};
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[2231]Largest Number After Digit Swaps by Parity.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int largestInteger(int num) {
        char[] digits = Integer.toString(num).toCharArray();
        PriorityQueue<Integer> even = new PriorityQueue<>((a, b) -> b - a);
        PriorityQueue<Integer> odd = new PriorityQueue<>((a, b) -> b - a);

        for (char c : digits) {
            int n = c - '0';

            if (n % 2 == 0) {
                even.add(n);
            } else {
                odd.add(n);
            }
        }

        for (int i = 0; i < digits.length; i++) {
            if ((digits[i] - '0') % 2 == 0) {
                digits[i] = (char) (even.poll() + '0');
            } else {
                digits[i] = (char) (odd.poll() + '0');
            }
        }

        return Integer.parseInt(new String(digits));
    }
}
//leetcode submit region end(Prohibit modification and deletion)
